docs/source/geopmgrid.1.rst

```diff
@@ -0,0 +1,341 @@
+    platform capabilities.
+
+--cpu-uncore-frequency CPU_UNCORE_FREQUENCY_DOMAIN  .. _cpu-uncore-frequency option:
+
+    Define a grid dimension over CPU_UNCORE_FREQUENCY_MAX_CONTROL for the
+    specified domain. Typically used with 'package' domain.
+
+--cpu-power CPU_POWER_DOMAIN  .. _cpu-power option:
+
+    Define a grid dimension over CPU_POWER_LIMIT_CONTROL for the specified
+    domain. Commonly used with 'package' or 'board' domains.
+
+--gpu-frequency GPU_FREQUENCY_DOMAIN  .. _gpu-frequency option:
+
+    Define a grid dimension over GPU_CORE_FREQUENCY_MAX_CONTROL for the
+    specified domain. Used with 'gpu' or 'gpu_chip' domains.
+
+--gpu-power GPU_POWER_DOMAIN  .. _gpu-power option:
+
+    Define a grid dimension over GPU_POWER_LIMIT_CONTROL for the specified
+    domain. The tool automatically detects whether to use Intel Level Zero
+    or NVIDIA NVML interfaces.
+
+--board-power BOARD_POWER_DOMAIN  .. _board-power option:
+
+    Define a grid dimension over BOARD_POWER_LIMIT_CONTROL for the specified
+    domain. Typically used with 'board' domain.
+
+Grid Navigation
+~~~~~~~~~~~~~~~
+
+--coordinate COORDINATE [COORDINATE ...]  .. _coordinate option:
+
+    Specify a coordinate within the grid to generate configuration for.
+    The number of coordinates must match the number of defined grid dimensions.
+    Each coordinate is an integer index into the corresponding dimension's
+    parameter range.
+
+--coordinate-file COORDINATE_FILE  .. _coordinate-file option:
+
+    Read grid coordinates from a file. The file should contain space-separated
+    integer coordinates corresponding to each grid dimension.
+
+--coordinate-range  .. _coordinate-range option:
+
+    Display the size of each dimension in the defined grid. This shows the
+    number of available parameter values for each control type and domain.
+
+Actions
+~~~~~~~
+
+--write  .. _write option:
+
+    Apply the configuration to the platform using the GEOPM service. Requires
+    either ``--coordinate`` or ``--coordinate-file`` to specify which grid
+    point to apply. This option pushes the control values directly to the
+    platform hardware.
+
+-h, --help  .. _help option:
+
+    Print help message and exit.
+
+
+Examples
+--------
+
+Exploring CPU frequency grid
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Define a grid over CPU frequency for all packages and display its dimensions:
+
+.. code-block:: shell-session
+
+   $ geopmgrid --cpu-frequency package --coordinate-range
+   Dimension 0: 21 points
+
+This shows that the CPU frequency grid has 21 available frequency settings
+across all package domains.
+
+Generating configuration commands
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generate geopmwrite commands for a specific grid point:
+
+.. code-block:: shell-session
+
+   $ geopmgrid --cpu-frequency package --coordinate 10
+   geopmwrite CPU_FREQUENCY_MAX_CONTROL package 0 2400000000
+   geopmwrite CPU_FREQUENCY_MAX_CONTROL package 1 2400000000
```

Comment:
Several of these can be reduced to 1 ``geopmwrite`` call at the board domain.

Reply:
In general these examples need to be run in the terminal to verify output